src/Http/Session/Session.php

```diff
@@ -0,0 +1,119 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Tempest\Http\Session;
+
+use const PHP_SESSION_ACTIVE;
+use RuntimeException;
+use SessionHandlerInterface;
+
+final class Session
+{
+    private bool $isStarted = false;
+
+    // TODO: allow the passing of options
+    public function __construct(private SessionHandlerInterface $sessionHandler)
+    {
+        // Here we just make sure that if the application ever
+        // prematurely stops (e.g., we call exit or die),
+        // the session data will still be saved.
+        session_register_shutdown();
+
+        $this->setSessionHandler($this->sessionHandler);
```

Comment:
Is setting the sessions handler something that needs to be accessed from the outside? When will people actually want to change their session handler while handling a request?

Reply:
When they want to seriously mess with their users. 🙃

For me, that was more of a consistency thing. The one scenario I can think of is if you ever wanted to migrate existing sessions to a new session handler (e.g., array to database).

Reply:
Ok. It's kind of an edge case, but I think a good enough reason to keep it as is